### ISSUE TITLE
PLW0128 skips assignment targets in square brackets and after asterisks

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#18660]
### Summary

[`redeclared-assigned-name` (PLW0128)](https://docs.astral.sh/ruff/rules/redeclared-assigned-name/) has false negatives for assignment targets in square brackets or after asterisks. It already correctly checks within parentheses.

```console
$ cat >plw0128.py <<'# EOF'
x, *x = 1, 2
[x, x] = 1, 2
# EOF

$ ruff --isolated check plw0128.py --select PLW0128
All checks passed!
```

### Version

ruff 0.11.13 (5faf72a4d 2025-06-05)
